pycbc/waveform/compress.py

```diff
@@ -749,18 +749,28 @@ def write_to_hdf(self, fp, template_hash, root=None):
             Put the `compressed_waveforms` group in the given directory in the
             hdf file. If `None`, `compressed_waveforms` will be the root
             directory.
+        precision : {None, str}
+            Cast the saved parameters to the given precision before saving. If
+            None provided, will use whatever their current precision is. This
+            will raise an error if the parameters have single precision but the
+            requested precision is double.
```

Comment:
Also, should we change `precision=args.precision` to `precision='double'` when doing `psd=pycbc.psd.from_cli(...)` because the original waveform and the decompressed waveform would have `dtype = numpy.complex128`? So, to be safe, wouldn't it be better to make all the three arrays ie. the two waveforms and the psd have the same precision?